src/engine/threaded_engine.cc

```diff
@@ -335,7 +335,9 @@ void ThreadedEngine::PushAsync(AsyncFn fn, Context exec_ctx,
 #endif
   ThreadedOpr *opr = NewOperator(std::move(fn), const_vars, mutable_vars, prop, opr_name, wait);
   opr->temporary = true;
-  const bool profiling = profiler_->IsProfiling(profiler::Profiler::kImperative);
```

Comment:
Previously this would check if it's imperative. Now you've extended the scope to Symbolic as well.
Could you explain why? Would be great to either have this commented in code or PR description